tests/mve/gtest_scene.cc

```diff
@@ -0,0 +1,751 @@
+    fs::mkdir(views_directory.c_str());
+
+    for (std::size_t i = 0; i < view_count; ++i)
+    {
+        const std::string view_directory_path = [&i, &views_directory](){
+            std::stringstream stream;
+            stream << fs::join_path(views_directory, "view_")
+                   << std::setw(4) << std::setfill('0') << std::to_string(i)
+                   << ".mve";
+            return stream.str();
+        }();
+        fs::mkdir(view_directory_path.c_str());
+        mve::View::Ptr view = mve::View::create();
+        view->set_name("view" + std::to_string(i));
+        view->set_id(i);
+        view->save_view_as(view_directory_path);
+    }
+
+    if(bundle != nullptr)
+        mve::save_mve_bundle(bundle, bundle_file);
+
+    return scene_directory;
+}
+
+void make_dirty(mve::View::Ptr view)
+{
+    view->set_name(view->get_name() + 'a');
+    assert(view->is_dirty());
+}
+
+void make_a_clean_view_dirty(mve::Scene::Ptr scene)
+{
+    using ViewList = mve::Scene::ViewList;
+    ViewList& views = scene->get_views();
+    ViewList::iterator clean_view = std::find_if(views.begin(), views.end(),
+        [](mve::View::Ptr& view) { return !view->is_dirty(); });
+    make_dirty(*clean_view);
+}
+
+mve::Scene::ViewList
+load_views_directly_from(const std::string& scene_directory)
+{
+    std::string view_directory = util::fs::join_path(scene_directory, "views");
+    util::fs::Directory directory(view_directory);
+    mve::Scene::ViewList list;
+    list.reserve(directory.size());
+
+    for (util::fs::File& file : directory)
+        list.push_back(mve::View::create(file.get_absolute_name()));
+
+    return list;
+}
+
+mve::Bundle::Ptr
+load_bundle_directly_from(const std::string& scene_directory)
+{
+    std::string bundle_file =
+            util::fs::join_path(scene_directory, "synth_0.out");
+    return mve::load_mve_bundle(bundle_file);
+}
+
+mve::Bundle::Ptr
+make_bundle(const std::size_t camera_count)
+{
+    mve::Bundle::Ptr bundle = mve::Bundle::create();
+    bundle->get_cameras().reserve(camera_count);
+    for(size_t i = 1; i <= camera_count; ++i)
+    {
+        mve::CameraInfo camera;
+        camera.flen = 1.0f + 2.0f / i
+                    * (i % 3 == 1); // one in every 3 cameras is not valid
+        camera.trans[0] = i - 10.0f;
+        camera.trans[1] = 1.0f / i;
+        camera.trans[2] = 10.0f - i;
+        camera.paspect = 0.5f + 1.0f / i;
+        bundle->get_cameras().push_back(std::move(camera));
+    }
+
+    return bundle;
+}
+
+bool
+views_match(mve::Scene::ViewList& lhs, mve::Scene::ViewList& rhs)
+{
+    if (lhs.size() != rhs.size())
+        return false;
+
+    for(mve::View::Ptr& left_view : lhs)
+    {
+        mve::Scene::ViewList::iterator found = std::find_if(
+                rhs.begin(),
+                rhs.end(),
+                [&left_view](mve::View::Ptr& right_view) {
+                    return left_view->get_id() == right_view->get_id();
```

Comment:
If it improves readability considerably, you can. But here I'd argue it doesn't.